library/runtime-ctrl-api/src/commonMain/kotlin/io/matthewnelson/kmp/tor/runtime/ctrl/api/address/LocalHost.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2024 Matthew Nelson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.matthewnelson.kmp.tor.runtime.ctrl.api.address
+
+import io.matthewnelson.kmp.file.IOException
+import io.matthewnelson.kmp.file.wrapIOException
+import io.matthewnelson.kmp.tor.core.api.annotation.InternalKmpTorApi
+import io.matthewnelson.kmp.tor.runtime.ctrl.api.internal.platformResolveIPv4
+import io.matthewnelson.kmp.tor.runtime.ctrl.api.internal.platformResolveIPv6
+import kotlin.concurrent.Volatile
+import kotlin.jvm.JvmStatic
+import kotlin.jvm.JvmSynthetic
+import kotlin.time.TimeSource
+
+public object LocalHost: Address("localhost") {
+
+    @JvmStatic
+    @Throws(IOException::class)
+    public fun resolveIPv4(): IPAddress.V4 = Cache.IPv4.resolve(checkCache = true)
+
+    @JvmStatic
+    @Throws(IOException::class)
+    public fun resolveIPv6(): IPAddress.V6 = Cache.IPv6.resolve(checkCache = true)
+
+    public override fun canonicalHostname(): String = value
+
+    private class Cache private constructor(private val address: IPAddress) {
+
+        private val timeMark = TimeSource.Monotonic.markNow()
+
+        private fun isExpired(): Boolean {
```

Comment:
Use `isNotExpired`, or `isStillValid` (something in the affirmative) and switch to `elapsedNanos < 5_000_000_250L`

Will be simpler to grok in associated `getOrNull()` function.